packages/examples/usage-tracking/usage-tracking.js

```diff
@@ -0,0 +1,106 @@
+// Import the Fixie client
+import { FixieClient } from 'fixie';
+import 'dotenv/config';
+import axios from 'axios';
+import { encode } from 'gpt-3-encoder';
+
+// Set the API key and Create the Fixie client
+const FIXIE_API_KEY = process.env.FIXIE_API_KEY;
+
+// Set our Agent ID that we want to get usage info on
+const AGENT_ID = process.env.FIXIE_AGENT_ID;
+
+// const fixieClient = new FixieClient({ apiKey: FIXIE_API_KEY });  // TODO once we have the right method
+
+let numConversations = 0;
+let numMessages = 0;
+let lenConversations = 0;
+let numTokens = 0;
+
+console.log('calling listAgentConversations');
+getAgentConversations(AGENT_ID).then((data) => {
+  numConversations = data.length;
+
+  console.log(`Got back ${data.length} conversations`);
+  data.forEach((element) => {
+    console.log('------------------------------------------------------------\n');
+    console.log(`Conversation ID: ${element.id}`);
+    console.log(`Total conversation turns\t${element.turns.length}`);
+
+    getConvoTurns(element);
+
+    // We want to get content for any messages that are:
+    // conversation.role = "user" -> get messages that are messages.kind = "text" and get messages.content for the text of the message
+    //      make sure the state is "done"
+    // conversation.role = "assistant"
+    //      kind = "text", get the content
+    //      make sure the state is "done"
+    //
+    //      kind = "functionResponse"....get the response
+  });
+
+  console.log('\n\n============================================================');
+  console.log(`Final stats for agent ${AGENT_ID}:\n`);
+  console.log(`Total Conversations\t${numConversations}\n`);
+  console.log(`Total Agent Messages\t${numMessages}\n`);
+  console.log(`Total Characters\t${lenConversations}\n`);
+  console.log(`Total LLM Tokens\t${numTokens}\n`);
+  console.log('============================================================');
+});
+
+function getConvoTurns(conversation) {
+  let numConvoMessages = 0;
+  let numConvoChars = 0;
+  let numConvoTokens = 0;
+
+  // Iterate thru the conversation and process all the turns
+  conversation.turns.forEach((turn) => {
+    numConvoMessages += turn.messages.length;
+
+    // Iterate through the turn messages and log their stats
+    turn.messages.forEach((message) => {
+      // console.log(message);
+      numMessages++;
+      // Make sure it's a message type that we want
+      if (turn.state == 'done') {
+        // Function Response
+        if (turn.role == 'assistant' && message.kind == 'functionResponse') {
+          lenConversations += message.response.length;
+          const encoded = encode(message.response);
+          numTokens += encoded.length;
+
+          numConvoChars += message.response.length;
+          numConvoTokens += encoded.length;
```

Comment:
This ignores the quadratic effect where messages become input tokens to subsequent generations (e.g. a conversation with 10 messages uses >> 10 times the number of input tokens as a conversation with 1 message)

But old messages also get pruned as well as they fill up the context window, so calculating it accurately is even trickier. The easiest thing to get approximately correct is to calculate the number of _output_ tokens, because that's just the assistant text/functionCall messages, but I'm guessing you want an estimate for both :)